### ISSUE TITLE
feasable to add spectator wallhack/esp for public games?

### ISSUE THREAD

**Issue by Automatics** [ValveSoftware/halflife#2504]
probably not possible to have it as good as csgo, but anything would be nice.

**Comment by Maxi605** [comment]
Would be amazing if its possible.

**Comment by CS-PRO1** [comment]
Will be great for all Goldsrc games if possible.

**Comment by SamVanheer** [comment]
Could you provide a video that shows this?

**Comment by basuritashka** [comment]
@SamVanheer 1:14 - https://www.youtube.com/watch?v=_CfpszI-k9U

**Comment by SamVanheer** [comment]
That effect is done using shaders, i don't think that's even possible to do in immediate mode, and how much of a performance hit there would be if it were possible. Regardless, changes would have to be made to networking code to send player positions at all times which can cause problems of its own.

**Comment by BlackShadow306** [comment]
![cs-4](https://user-images.githubusercontent.com/30868395/59315792-f0ebb000-8cc3-11e9-8c36-36adaac28fb8.jpg)

I believe this can be implemented like the ESP Wallhack effect people did. But still not worth investing time.

**Comment by RauliTop** [comment]
Not necessary.
We have other problems or new features better.

Don't try to change CS 1.6 into CSGO, it's a bad idea

**Comment by Automatics** [comment]
yes, we shouldnt have anti cheat tools useful to 99% of the playerbase, makes sense.

**Comment by Maxi605** [comment]
> Not necessary.
> We have other problems or new features better.
> 
> Don't try to change CS 1.6 into CSGO, it's a bad idea

Well, the anticheat isn't really useful, there aren't any third party anticheats to help, the game is old and cheats are always going to be there but giving the server providers tools to combat cheating isn't a bad idea and i don't see how is turning CS 1.6 to CS:GO, is just a "simple" idea that would be really helpful.

**Comment by SamVanheer** [comment]
This is far from simple though, it represents a considerable amount of effort in re-engineering model rendering code and player networking.

**Comment by Maxi605** [comment]
> This is far from simple though, it represents a considerable amount of effort in re-engineering model rendering code and player networking.

Yeah, i know, that's why i put "simple", i know it isn't, it WOULD be a great addition to the game, it isn't really need it right now but if it at least gets considered for the future i'm happy

**Comment by Automatics** [comment]
keep in mind i didnt ask for that, someone changed the title, id be happy with a pixel representing player position behind wall.

**Comment by ChaosSpoofer** [comment]
> This is far from simple though, it represents a considerable amount of effort in re-engineering model rendering code and player networking.

Not really.  The reason why people want this feature is to watch demos and consider whether the player is wallhacking or not.  Wallhacks _already_ operate in GoldSrc by showing what's within render range, typically only behind a wall or two at most, rather than across the whole map.  An admin tool, matching what's already possible, is precisely what admins need to expose and deal with cheaters.  Historically, such tools have been available via third-party programs, though with each new update, fewer and fewer still work.

Completely overhauling the code to allow x-ray throughout the entire map would instead be counterproductive, as that would mainly serve to strengthen wallhacks to begin with.  But it would certainly be helpful to allow admins (or players in general) the ability to replicate existing wallhack behavior while reviewing demos for cheat detection purposes.
